server/mailet/integration-testing/src/test/java/org/apache/james/smtp/SmtpRandomStoringTest.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.james.core.builder.MimeMessageBuilder;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailets.TemporaryJamesServer;
+import org.apache.james.mailets.configuration.CommonProcessors;
+import org.apache.james.mailets.configuration.MailetConfiguration;
+import org.apache.james.mailets.configuration.MailetContainer;
+import org.apache.james.mailets.configuration.ProcessorConfiguration;
+import org.apache.james.mailets.configuration.SmtpConfiguration;
+import org.apache.james.modules.MailboxProbeImpl;
+import org.apache.james.modules.protocols.ImapGuiceProbe;
+import org.apache.james.modules.protocols.SmtpGuiceProbe;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.server.core.MailImpl;
+import org.apache.james.transport.mailets.RandomStoring;
+import org.apache.james.transport.matchers.All;
+import org.apache.james.transport.matchers.SMTPIsAuthNetwork;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.FakeSmtp;
+import org.apache.james.utils.IMAPMessageReader;
+import org.apache.james.utils.SMTPMessageSender;
+import org.apache.mailet.Mail;
+
+import org.awaitility.Duration;
+import org.awaitility.core.ConditionFactory;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+public class SmtpRandomStoringTest {
+    private static final String FROM = "from@" + DEFAULT_DOMAIN;
+    private static final String TO = "to@any.com";
+    private static final Long USERS_NUMBERS = 10L;
+    private static final ConditionFactory awaitAtMostTenSeconds = calmlyAwait
+        .atMost(new Duration(10L, TimeUnit.SECONDS));
+
+    private static final ImmutableList<String> USERS = LongStream.range(1L, USERS_NUMBERS)
+        .boxed()
+        .map(index -> "user" + index + "@" + DEFAULT_DOMAIN)
+        .collect(Guavate.toImmutableList());
+
+    private static final MailetConfiguration RANDOM_STORING = MailetConfiguration.builder()
+            .matcher(All.class)
+            .mailet(RandomStoring.class)
+            .build();
+
+    @ClassRule
+    public static FakeSmtp fakeSmtp = new FakeSmtp();
+    @Rule
+    public IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private TemporaryJamesServer jamesServer;
+    private ImapGuiceProbe imapProbe;
+
+    @BeforeClass
+    public static void setup() {
+        fakeSmtp.awaitStarted(awaitAtMostOneMinute);
+    }
+
+    private void createJamesServer(SmtpConfiguration.Builder smtpConfiguration) throws Exception {
```

Comment:
This method has a lot of outputs and does more than its title.
Could we split it ?
What prevents from calling it in a @Before method ?